src/qt/anybutton.cpp

```diff
@@ -24,18 +25,87 @@ class wxQtPushButton : public wxQtEventSignalHandler< QPushButton, wxAnyButton >
 {
 
 public:
+    virtual bool eventFilter(QObject *watched, QEvent *event);
     wxQtPushButton( wxWindow *parent, wxAnyButton *handler);
 
 private:
     void clicked(bool);
+    wxAnyButton *m_button;
 };
 
 wxQtPushButton::wxQtPushButton(wxWindow *parent, wxAnyButton *handler)
     : wxQtEventSignalHandler< QPushButton, wxAnyButton >( parent, handler )
 {
+    m_button = handler;
+    installEventFilter( this );
     connect(this, &QPushButton::clicked, this, &wxQtPushButton::clicked);
 }
 
+bool wxQtPushButton::eventFilter(QObject *watched, QEvent *event)
+{
+// TODO: Disabled control shouldn't get any events
+    QPushButton *button = qobject_cast<QPushButton *>( watched );
+    if( !button )
+        return false;
+    QIcon icon = button->icon();
+    if( event->type() == QEvent::HoverMove && button->isEnabled() )
+    {
+        if( m_button->GetStateBitmaps()[1].IsOk() )
+        {
+            button->setIcon( QIcon( *( m_button->GetStateBitmaps()[1].GetHandle() ) ) );
+//            return true;
```

Comment:
Commented code should not be part of the [new] code.

Reply:
Will remove in the next post